### PR TITLE
fix: scientific notation of numbers parsed incorrectly

### PR DESCRIPTION
**Pull request by florissmit10**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
This PR adresses issue [#6314](https://github.com/risingwavelabs/risingwave/issues/6314).
After this change users can use scientific numbers in select and insert statements.

Scientific numbers should be usable in select and insert statements.
- [x] Scientific numbers can be tokenized/parsed (1e6, 1e-5, 1.25e5)
- [x] Parsed scientific numbers can be processed in select statement
- [x] Parsed scientific numbers can be processed in insert statement 
- [x] add e2e test

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Release note

It is now possible to use scientific number notation (1e6, 1.25e5, 1e-4) in select and insert queries.

### CODE REVIEW

**Comment by TennyZhuang** [comment]
Thanks for your contribution!

If the PR is ready to review, feel free to mark it as ready and ask me or @xiangjinwu for a review.

**Review by wangrunji0408 — COMMENTED** [review]
Thanks for your contribution! Generally LGTM.

File: e2e_test/batch/basic/scientific_number.slt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
+
+query IIRR
+SELECT 1e6, -1e6, 1e-6, -1e-6, -1.25e3, 1.25e3
```
Comment (lmatz) [inline e2e_test/batch/basic/scientific_number.slt:5]: Thanks for your contribution!
We can add some test cases where the number overflows or underflows and thus returns an error,
but we can also do it later

**Comment by florissmit10** [comment]
Thanks for the feedback everyone! 

Right now using a scientific number always returns a Decimal, is that ok?

**Review by wangrunji0408 — APPROVED** [review]
Looks cool!

**Comment by xiangjinwu** [comment]
> Right now using a scientific number always returns a Decimal, is that ok?

According to Additional Context [link](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-CONSTANTS-NUMERIC) in this issue
> A numeric constant that contains neither a decimal point nor an exponent is initially presumed to be type `integer` if its value fits in type `integer` (32 bits); otherwise it is presumed to be type `bigint` if its value fits in type `bigint` (64 bits); otherwise it is taken to be type `numeric`. **Constants that contain decimal points and/or exponents are always initially presumed to be type `numeric`.**

**Pull request merged**